research-software-directory/RSD-as-a-service
# SURFconext auth module and nginx integration

# SURFconext auth module and nginx integration

Fixes #50 #8 

The implementation of SURFConext authentication on the front-end and the integration with nginx as reverse proxy.

How to test:

* `docker-compose down --volumes`: to remove old containers
* `docker-compose build`: to build new images
* `docker-compose up`: to start the solution. Examine the logs: the test should be passing for fronend and auth-tests
* The app should be visible on http://localhost
* Test data migration module by following these steps: 
```bash
# navigate to data-migration folder
cd data-migration
# run docker-compose
docker-compose up
```
* Test login with SURFconext:
    * First navigate to [software page](http://localhost/software). Note! you should be redirected back to this page after login process is completed
    * click on SingIn button in the top right corner of the screen 
![image](https://user-images.githubusercontent.com/9204081/148824054-9ad9d666-4a00-4913-a152-97e3035c3561.png)
    * select SURFconext Test IdP and login with credentials: professor1 / professor1
    * you should be redirect back to software page and the avatar should be visible in the right top corner 
![image](https://user-images.githubusercontent.com/9204081/148824518-99648e7f-565b-4654-b7ec-20bd05a206ac.png)
    
*  Test refresh tokens with SURFconext: navigate to my profile page using menu at user avatar. Note the exp value and wait for about 1 minute. The exp value will change. You can also examine browser console logs which should have logs about refresh schedule. Note! The token expiration time is currently set to very low value in order to test token refresh is working properly. The token expiration time should be 

## Nginx setup

The reverse proxy is setup with this routes:
* [root](http://localhost): home page of RSD (frontend)
* [auth](http://localhost/auth): authentication module (Java)
* [api/v1](http://localhost/api/v1): postgREST backend service (Haskel)
* [api/fe](http://localhost/api/fe): api service of frontend (currently citations download and refresh tokens endpoint) (NodeJS)
